### PR TITLE
[CONFIG] Image upload service

### PR DESCRIPTION
**Pull request by marce-alf**
## Jira Ticket

- [WA-32]

## Description
Created image upload service so users can upload and update their avatar image.
The valid image extensions are (PNG/JPG/SVG/WEBP).

AS: a user
I Want: to upload and update my avatar image
To: keep the data updated.

## Evidence:
Multer validator when providing a valid image and valid user values
avatar-ok

Multer validator when providing an invalid image
avatar-wrong-image

### CODE REVIEW

**Review by HugoCavalero92 — APPROVED** [review]

File: middlewares/multer/avatarUpload.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    return cb(new ErrorObject("Invalid avatar file extension.", 422));
+  },
+  storage: ds,
+});
+const uploadAvatar = multerConfig.single("avatar");
+
+module.exports = (req, res, next) => {
+  uploadAvatar(req, res, (error) => {
+    if (error) return res.status(error.statusCode).json(error);
```
Comment (ledelaespriella) [inline middlewares/multer/avatarUpload.js:33]: error in this line: 

whitout code:

image

image

whit code:

image

File: middlewares/multer/avatarUpload.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+const { ErrorObject } = require("../../helpers/error");
+const multer = require("multer");
+const ds = multer.diskStorage({
+  destination: (req, file, cb) => {
+    cb(null, "public/avatars");
+  },
+  filename: (req, file, cb) => {
+    req.body.avatar = `http://localhost:${process.env.PORT || 3000}/avatars/${
```
Comment (ledelaespriella) [inline middlewares/multer/avatarUpload.js:8]: this route is not implemented

**Review by ledelaespriella — CHANGES_REQUESTED** [review]

File: middlewares/multer/avatarUpload.js (hunk 2 above)
Comment (marce-alf) [reply middlewares/multer/avatarUpload.js:8]: express.static(path.join(__dirname, "public"))
This is a route created and exposed by express.static, you can find it in app.js.

**Review by ledelaespriella — CHANGES_REQUESTED** [review]
resolve conflicts

**Review by ledelaespriella — APPROVED** [review]

**Pull request merged**